### ISSUE TITLE
Comskip occasionally not running for iTunes exports

### ISSUE THREAD

**Issue by Brandonshire** [essandess/etv-comskip#65]
It's possible this is what the new version addresses, but I've had several recordings lately (frustratingly not ALL of them, they seem to be failing somewhat randomly) that didn't automatically export to iTunes with commercials marked. Luckily they didn't cut commercials out at all but they were also not marked as chapters. I'm trying to find some commonality between them, but so far I'm not finding anything. Some that had them marked and some that didn't were even recorded on the same night, and on the same channel! (though not at the same or overlapping times). I'll keep an eye on it and see if I can figure out a pattern or if it keeps happening. Hopefully maybe the new version fixes this issue somehow?

Just wanted to let you know. And thanks for this project. I really appreciate all the work you've put into it.

**Comment by Brandonshire** [comment]
Ok, just noticed that one of the recordings that exported without commercials marked doesn't seem to have them marked in the EyeTV recording either. The others that exported that way do, but that explains that one.

**Comment by essandess** [comment]
It sounds like the same issue that the latest v3.4.1 should address. I also had this problem very occasionally and tracked  it down to iTunes returning `missing value`'s in a return list to a query request. I've tested this for a week and believe it's addressed but we're honestly down to rare event issues now.

Please uninstall/install and watch for the issue. The relevant logs are in:

```
~/Library/Logs/EyeTV\ scripts.log
~/Library/Logs/ETVComskip/<recording_id>*.log
<EyeTV archive>/*.eyetv/<recording_id>.log
```

**Comment by Brandonshire** [comment]
Thanks, that's the answer I was hoping for! I'll keep an eye on it. (And yes, I'm glad we're getting to the point of dealing with edge cases now!).

**Comment by Brandonshire** [comment]
So I was out most of this weekend but ran several test recordings (as well as some regularly scheduled recordings). Unfortunately in at least two cases (out of 5 recordings I think) commercials were not in the automated export to iTunes. In at least one case commercials weren't marked at all (not in EyeTV not in the export). And of course in several cases everything worked exactly as expected. Still no obvious pattern that I can see in which it works with and which it doesn't. I'll hopefully have some time tonight or tomorrow night to try to get more detail. Just wanted to let you know...

**Comment by essandess** [comment]
Thanks for looking into this. Please check the logs for something to pin blame on, etv-comskip or other. I've been running without issue so I'll need pointers.

**Comment by Brandonshire** [comment]
Ok, had a few minutes this morning and checked on a recording from last night. No commercials were marked in EyeTV or in the export to iTunes.

Unfortunately I only had enough time to grab one of the logs you mentioned. I'll get the rest tonight but this one appears to be filled with errors, so it may be enough...

Here is the EyeTV scripts.log for that recording:

12/7/15 7:59:10 PMRecordingStarted run for ID: 471229140
12/7/15 9:02:00 PM RecordingDone run for ID: 471229140
12/7/15 9:43:53 PM ExportDone run for ID: 471229140
	ExportDone::Title: Supergirl
	Location: /Users/stephenb/Documents/EyeTV Archive/Supergirl - Human for a Day.eyetv/000000001c1662d4.eyetvr
	ExportDone::repeatloop mytv error: iTunes got an error: CanÕt get location of every track of playlist "TV Shows" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mymovies error: iTunes got an error: CanÕt get location of every track of playlist "Movies" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mytv error: iTunes got an error: CanÕt get location of every track of playlist "TV Shows" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mymovies error: iTunes got an error: CanÕt get location of every track of playlist "Movies" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mytv error: iTunes got an error: CanÕt get location of every track of playlist "TV Shows" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mymovies error: iTunes got an error: CanÕt get location of every track of playlist "Movies" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mytv error: iTunes got an error: CanÕt get location of every track of playlist "TV Shows" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mymovies error: iTunes got an error: CanÕt get location of every track of playlist "Movies" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::empty mytv!!!
12/7/15 10:57:40 PM ExportDone run for ID: 471229140
	ExportDone::Title: Supergirl
	Location: /Users/stephenb/Documents/EyeTV Archive/Supergirl - Human for a Day.eyetv/000000001c1662d4.eyetvr
	ExportDone::repeatloop mytv error: iTunes got an error: CanÕt get location of every track of playlist "TV Shows" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mymovies error: iTunes got an error: CanÕt get location of every track of playlist "Movies" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mytv error: iTunes got an error: CanÕt get location of every track of playlist "TV Shows" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mymovies error: iTunes got an error: CanÕt get location of every track of playlist "Movies" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mytv error: iTunes got an error: CanÕt get location of every track of playlist "TV Shows" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mymovies error: iTunes got an error: CanÕt get location of every track of playlist "Movies" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mytv error: iTunes got an error: CanÕt get location of every track of playlist "TV Shows" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::repeatloop mymovies error: iTunes got an error: CanÕt get location of every track of playlist "Movies" whose name of it = "Supergirl" or artist of it = "Supergirl".; error number -1728.
	ExportDone::empty mytv!!!

**Comment by essandess** [comment]
Those errors may be ok. The EyeTV AppleScript dictionary doesn't communicate whether the exports are exports to a MP4 format (within the `.eyetv` directory) or an iTunes export, so the ExportDone script assumes there's an iTunes export and searches for it. If it doesn't exist, the search fails as above. Was this an iTunes export that ExportDone should've found?

Other possibilities are non-responsiveness of the EyeTV app to calls from the triggered scripts or `MarkCommercials`, or a mismatch in the iTunes query term.

**Comment by Brandonshire** [comment]
It was a recording that was automatically exported to iTunes. This was the
relevant part of that particular log I found this morning. Tonight I'll try
to get the other logs you mentioned above, and maybe they'll shed more
light on what's happening.

**Comment by essandess** [comment]
Sounds like a mismatch in the query terms[?] If so, there may not be an easy fix, as these are set by EyeTV, and its AppleScript dictionary doesn't provide the export's title. When I have a moment I'll figure out how to have EyeTV report its metadata and maybe a solution will present itself. The current query looks like:

```
tell application "EyeTV"
	set myshortname to get the title of recording id myid
end tell
tell application "iTunes"
	set mytv to get the location of (the tracks of playlist "TV Shows" whose name is myshortname or artist is myshortname)
end tell
```

If that query doesn't yield any iTunes recordings, then commercial markings cannot be exported to iTunes.

**Comment by Brandonshire** [comment]
So I'm looking and I can't seem to find any of the other logs you ask for for this recording.

Again I want to point out that, in this case, no commercials were marked within the EyeTV recording either. They're not just missing in the export to iTunes.

**Comment by essandess** [comment]
Hmmm… looks like it's necessary to add more verbosity to the triggered script `DEBUG` statements to get more information. To summarize, in this instance neither `MarkCommercials` was run (so no logs), nor `ExportDone` discovered the iTunes export, correct?

**Comment by Brandonshire** [comment]
That is correct.

So, we might be dealing with two different problems here (ideally they're related and might have one solution but... I guess we'll see). I have another recording from Sunday in which commercials were marked within EyeTV, but not in the export. I'll post the logs for that momentarily.

**Comment by Brandonshire** [comment]
Here is the log from: ~/Library/Logs/EyeTV\ scripts.log

12/6/15 9:20:53 PM ExportDone run for ID: 471144540
	ExportDone::Title: Brooklyn Nine-Nine
	Location: /Users/stephenb/Documents/EyeTV Archive/Brooklyn Nine-Nine - The Swedes.eyetv/000000001c15185c.eyetvr
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:New Captain.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:The Funeral.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:Boyle's Hunch.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:The Oolong Slayer.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:Halloween III.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:Into the Woods.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:The Mattress.m4v
	ExportDone::mymp4: /Volumes/Fist of Fury/iTunes Library/TV Shows/Brooklyn Nine-Nine/Season 3/The Mattress.m4v
12/6/15 9:36:59 PM ExportDone run for ID: 471144540
	ExportDone::Title: Brooklyn Nine-Nine
	Location: /Users/stephenb/Documents/EyeTV Archive/Brooklyn Nine-Nine - The Swedes.eyetv/000000001c15185c.eyetvr
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:New Captain.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:The Funeral.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:Boyle's Hunch.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:The Oolong Slayer.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:Halloween III.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:Into the Woods.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:The Mattress.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Brooklyn Nine-Nine:Season 3:The Swedes.m4v
	ExportDone::mymp4: /Volumes/Fist of Fury/iTunes Library/TV Shows/Brooklyn Nine-Nine/Season 3/The Mattress.m4v

**Comment by Brandonshire** [comment]
Here is the log (for the same recording as above from: ~/Library/Logs/ETVComskip/<recording_id>*.log

Sun Dec  6 21:37:29 2015 - /Library/Application Support/ETVComskip/bin/MarkCommercials, 1.1 starting
Sun Dec  6 21:37:29 2015 - Command line: ['/Library/Application Support/ETVComskip/bin/MarkCommercials', '--log', '471144540']
Sun Dec  6 21:37:29 2015 - Application name: EyeTV
Sun Dec  6 21:37:30 2015 - Cfg files read: []
Sun Dec  6 21:37:30 2015 - Checking communications to EyeTV with 3 retries
Sun Dec  6 21:37:30 2015 -   Attempt 1 worked
Sun Dec  6 21:37:30 2015 - Getting recordings
Sun Dec  6 21:37:30 2015 - Recordings: [app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470537940), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470543340), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471104940), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471146340), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469832340), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(468721740), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(466909140), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469326540), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469934940), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469330140), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471144540), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(468899940), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470714340), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469504740), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(436845541), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(468989940), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470804340), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469594740), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469831862), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(468809940), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469414740), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470624340), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469420439), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470637240), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470723640), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470810040), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(470896440), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(468723540), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(469933140), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471142740), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471029340), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471034911), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471038340), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471040140), app(u'/Applications/EyeTV/EyeTV.app').recordings.ID(471036540)]
Sun Dec  6 21:37:30 2015 - Found recording 471144540
Sun Dec  6 21:37:30 2015 - Marker count: 0
Sun Dec  6 21:37:30 2015 -  1. Processing "Brooklyn Nine-Nine" on [WTTG-DT] channel [5-1]...
Sun Dec  6 21:37:30 2015 - Path to recording is /Users/stephenb/Documents/EyeTV Archive/Brooklyn Nine-Nine - The Swedes.eyetv/000000001c15185c.eyetvr
Sun Dec  6 21:37:30 2015 - Changing directory to /Users/stephenb/Documents/EyeTV Archive/Brooklyn Nine-Nine - The Swedes.eyetv
Sun Dec  6 21:37:30 2015 - Running: /usr/bin/nice -n 14 /Library/Application Support/ETVComskip/bin/gtimeout 5h /Library/Application Support/ETVComskip/bin/comskip --ini=/Library/Application Support/ETVComskip/comskip.ini 000000001c15185c.mpg
Sun Dec  6 21:41:07 2015 - Return code is: 0, 0x0
Sun Dec  6 21:41:07 2015 - Error code is: 0, 0x0
Sun Dec  6 21:41:07 2015 - Commercials found by comskip
Sun Dec  6 21:41:07 2015 - Plist file contents: ['208.42\t500.52\t0\n', '732.28\t912.86\t0\n', '1405.97\t1521.79\t0\n', '1857.26\t1979.43\t0\n']
Sun Dec  6 21:41:07 2015 - Processing plist file line: '208.42	500.52	0
'
Sun Dec  6 21:41:07 2015 - Adding marker, start: 208, end: 500
Sun Dec  6 21:41:07 2015 - Processing plist file line: '732.28	912.86	0
'
Sun Dec  6 21:41:07 2015 - Adding marker, start: 732, end: 912
Sun Dec  6 21:41:07 2015 - Processing plist file line: '1405.97	1521.79	0
'
Sun Dec  6 21:41:07 2015 - Adding marker, start: 1405, end: 1521
Sun Dec  6 21:41:07 2015 - Processing plist file line: '1857.26	1979.43	0
'
Sun Dec  6 21:41:07 2015 - Adding marker, start: 1857, end: 1979
Sun Dec  6 21:41:07 2015 - Setting markers on recording
Sun Dec  6 21:41:07 2015 - Adding marker: [{'position': 208.42, aem.AEType('leng'): 292.1}, {'position': 732.28, aem.AEType('leng'): 180.58000000000004}, {'position': 1405.97, aem.AEType('leng'): 115.81999999999994}, {'position': 1857.26, aem.AEType('leng'): 122.17000000000007}]
Sun Dec  6 21:41:07 2015 - Exiting

**Comment by Brandonshire** [comment]
And I'm having trouble finding the last one... (<EyeTV archive>/*.eyetv/<recording_id>.log) Is the attached what you're looking for? (converted to a text file so that github would accept it as an attachment)
[000000001c15185c.txt](https://github.com/essandess/etv-comskip/files/56201/000000001c15185c.txt)

**Comment by essandess** [comment]
The last line of `ExportDone` says it found this iTunes export:

```
ExportDone::mymp4: /Volumes/Fist of Fury/iTunes Library/TV Shows/Brooklyn Nine-Nine/Season 3/The Mattress.m4v
```

But the `MarkCommercials` log says it marked this show:

```
Sun Dec 6 21:37:30 2015 - Changing directory to /Users/stephenb/Documents/EyeTV Archive/Brooklyn Nine-Nine - The Swedes.eyetv
```

Which is correct? Assuming that it's the Swedes, would you please confirm that the Mattress iTunes export was incorrectly mark d with the Swedes export?

**Comment by Brandonshire** [comment]
The Swedes is the correct recording for this session. (The Mattress is an older episode from this season).

**Comment by Brandonshire** [comment]
How would I confirm that? Just look at when the commercials are marked in it and see if they line up?

**Comment by essandess** [comment]
Thanks for that last comskip log. All I really need to know is that comskip's `.edl` and `.log` file exist.

**Comment by Brandonshire** [comment]
Ok, just checked and yes it looks like the chapter markers on "The Mattress" line up with the ones that should have been on "The Swedes."

**Comment by Brandonshire** [comment]
Yes, there is both a .edl file and a .log file in the package for the episode in question (The Swedes).

**Comment by essandess** [comment]
This Swedes versus Matress looks like a etv-comskip bug. The code is intended to take the last recording that matches the query. Would you please find the `.m4v` files for each recording in the iTunes archive and report this:

```
ls -l Swedes.m4v Mattress.m4v
```

Of course you'll need full file names.

**Comment by Brandonshire** [comment]
Swedes:

-rw-r--r--@ 1 stephenb  staff  1262958664 Dec  6 21:36 /Volumes/Fist of Fury/iTunes Library/TV Shows/Brooklyn Nine-Nine/Season 3/The Swedes.m4v

Mattress:

-rw-r--r--@ 1 stephenb  staff  1270306575 Dec  6 21:41 /Volumes/Fist of Fury/iTunes Library/TV Shows/Brooklyn Nine-Nine/Season 3/The Mattress.m4v

**Comment by essandess** [comment]
How is that these shows are both exported five minutes apart? Did you export by hand?

**Comment by Brandonshire** [comment]
Nope. I'm guessing that the similar time is related to one exporting and the other being modified (via commercial marking) a few minutes late. The Mattress was originally added to iTunes (and recorded) on November 16th.

**Comment by essandess** [comment]
Yes, of course. This has to be a bug. I'll investigate and add a little more verbosity to the logs. Thanks again for your help and stay tuned…

```
/Volumes/Fist of Fury/
```

> ![jin wu marial arts academy](https://static.wixstatic.com/media/b8f782_b892450a85544b84b2c2509d9753b4e5.png/v1/fill/w_250,h_250,al_c,usm_0.50_1.20_0.00/b8f782_b892450a85544b84b2c2509d9753b4e5.png)

**Comment by Brandonshire** [comment]
(For a while I was naming my disks after Bruce Lee Movies, I have another volume named Enter The Dragon on my desktop right now!)

**Comment by Brandonshire** [comment]
It happened again (The same issue as the Supergirl episode. No commercial marking at all, in EyeTV or in the export). No log or edl file in the recording, and no etvcomskip log.

Here's the relevant section of the EyeTV log:

12/8/15 8:59:10 PMRecordingStarted run for ID: 471319140
12/8/15 10:02:01 PM RecordingDone run for ID: 471319140
12/8/15 10:42:31 PM ExportDone run for ID: 471319140
	ExportDone::Title: Marvel's Agents of S.H.I.E.L.D.
	Location: /Users/stephenb/Documents/EyeTV Archive/Marvel's Agents of S.H.I.E.L.D. - Maveth.eyetv/000000001c17c264.eyetvr
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Chaos Theory.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Closure.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Laws of Nature.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Purpose in the Machine.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:A Wanted (Inhu)man.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Devils You Know.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:05 4,722 Hours (HD).mp4
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:06 Among Us Hide... (1080p HD).mp4
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Many Heads, One Tale.m4v
	ExportDone::mymp4: /Volumes/Fist of Fury/iTunes Library/TV Shows/Marvel's Agents of S.H.I.E.L.D_/Closure.m4v
12/8/15 11:19:46 PM ExportDone run for ID: 471319140
	ExportDone::Title: Marvel's Agents of S.H.I.E.L.D.
	Location: /Users/stephenb/Documents/EyeTV Archive/Marvel's Agents of S.H.I.E.L.D. - Maveth.eyetv/000000001c17c264.eyetvr
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Chaos Theory.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Closure.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Laws of Nature.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Purpose in the Machine.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:A Wanted (Inhu)man.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Devils You Know.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:05 4,722 Hours (HD).mp4
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:06 Among Us Hide... (1080p HD).mp4
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Many Heads, One Tale.m4v
	ExportDone::repeatloop mytv: Fist of Fury:iTunes Library:TV Shows:Marvel's Agents of S.H.I.E.L.D_:Season 3:Maveth.m4v
	ExportDone::mymp4: /Volumes/Fist of Fury/iTunes Library/TV Shows/Marvel's Agents of S.H.I.E.L.D_/Closure.m4v

**Comment by essandess** [comment]
I've been looking for this issue and haven't seen it, so I've modified the [ExportDone](https://github.com/essandess/etv-comskip/blob/master/src/scripts/ExportDone.applescript) script with some timing and debug statements.

Would you please download the modified script from this [link](https://github.com/essandess/etv-comskip/blob/master/src/scripts/ExportDone.applescript), run the following commands, and look for a recording with the issue you describe?

```
osacompile -do ~/Downloads/ExportDone.scpt ~/Downloads/ExportDone.applescript
sudo cp ~/Downloads/ExportDone.scpt /Library/Application\ Support/EyeTV/Scripts/TriggeredScripts/ExportDone.scpt
# Restart EyeTV.app
```

It appears that the `IsFileOpen` occasionaly returns `true` for the latest iTunes export. To confirm, I've added a few debugging statements, and increased the number of minutes the script waits for iTunes to finish whatever bookkeeping and database updates it does.

At the very least, I'll add a timing test to ensure that older exports are not tagged as occurred above.

**Comment by Brandonshire** [comment]
So things have been busy lately (Mom's birthday, family visiting, holidays
coming up, an office move, etc.) so I haven't really had much time to look
in on this, I'll try to get this going soon and see what I can report.
Unfortunately it's gotten to the point now where it seems like the majority
of recordings dont' have commercials properly marked. Either they don't
have them marked at all (within the EyeTV archive or in iTunes) or they
have them in the EyeTV Archive but not in the iTunes export (which may mean
they were marked on another recording or another issue, I haven't had time
to delve into it too much to figure that out).

I think things may calm down a little soon, so I'll try to get you whatever
info I can when I can, but it will likely be this weekend at the earliest
before I can sit down and do this.

Thanks so much for your hard work, I hope I can help out soon!

-Stephen

**Comment by essandess** [comment]
In that case let me add a couple more improvements, then test after I post again. 

Also, would you please describe your box—make, year, CPUs, memory. I'm thinking this has something to do with the way iTunes handles huge imports. These are all very large Apple TV 1080p h.264's, correct?

**Comment by essandess** [comment]
Please give this a try with the latest [ExportDone](https://github.com/essandess/etv-comskip/blob/master/src/scripts/ExportDone.applescript) and [RecordingDone](https://github.com/essandess/etv-comskip/blob/master/src/scripts/RecordingDone.applescript) scripts and the commands:

```
osacompile -do ~/Downloads/ExportDone.scpt ~/Downloads/ExportDone.applescript
sudo cp ~/Downloads/ExportDone.scpt /Library/Application\ Support/EyeTV/Scripts/TriggeredScripts/ExportDone.scpt
osacompile -do ~/Downloads/RecordingDone.scpt ~/Downloads/RecordingDone.applescript
sudo cp ~/Downloads/RecordingDone.scpt /Library/Application\ Support/EyeTV/Scripts/TriggeredScripts/RecordingDone.scpt
# Restart EyeTV.app
```

**Comment by Brandonshire** [comment]
Hey there! The holidays (really the entire month of December) have been super busy for me so I'm sorry I've kind of disapeared. I haven't really been home much and when I have, playing with an optional (if very cool!) sub-system of my TV set up has been a low priority.

However! I wanted to report that a few weeks back I uninstalled, and reinstalled the (then) most recent build (I see a new one came out today), which I had already done once, and so far I haven't had any problems. I'm not sure what has changed in my set up, or if it has something to do with the shows that have been recording (I've set up some smart-lists for several shows I don't normally watch just so I have some thing to test with since most of my normal shows are on hiatus), but I haven't had a problem in a few weeks. That is to say that commercials are getting marked in both the EyeTV archives, and in the iTunes exports, and the commercials in the exports are going to the correct episodes.

I'll install the latest version and keep an eye on it, and when things calm down more (hopefully soon) I'll try to get you more info, but so far, for now, things are working (though I couldn't tell you why).
